beacon-chain/db/kv/operations.go

```diff
@@ -13,32 +13,23 @@ import (
 func (kv *Store) VoluntaryExit(ctx context.Context, exitRoot [32]byte) (*ethpb.VoluntaryExit, error) {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.VoluntaryExit")
 	defer span.End()
-	var exit *ethpb.VoluntaryExit
-	err := kv.db.View(func(tx *bolt.Tx) error {
-		bkt := tx.Bucket(voluntaryExitsBucket)
-		enc := bkt.Get(exitRoot[:])
-		if enc == nil {
-			return nil
-		}
-		exit = &ethpb.VoluntaryExit{}
-		return decode(ctx, enc, exit)
-	})
-	return exit, err
+	enc, err := kv.voluntaryExitBytes(ctx, exitRoot)
+	if err != nil {
+		return nil, err
+	}
+	exit := &ethpb.VoluntaryExit{}
+	return exit, decode(ctx, enc, exit)
 }
 
 // HasVoluntaryExit verifies if a voluntary exit is stored in the db by its signing root.
 func (kv *Store) HasVoluntaryExit(ctx context.Context, exitRoot [32]byte) bool {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.HasVoluntaryExit")
 	defer span.End()
-	exists := false
-	if err := kv.db.View(func(tx *bolt.Tx) error {
-		bkt := tx.Bucket(voluntaryExitsBucket)
-		exists = bkt.Get(exitRoot[:]) != nil
-		return nil
-	}); err != nil { // This view never returns an error, but we'll handle anyway for sanity.
+	enc, err := kv.voluntaryExitBytes(ctx, exitRoot)
+	if err != nil {
 		panic(err)
```

Comment:
why is this a panic ?

Reply:
What should it be instead? The previous comment details the rationale, it was handled the same way previously since the view would never error.

Reply:
I would rather we never have a panic for non critical code, but nvm lets just leave it for now.